### PR TITLE
Add triangle collisions

### PR DESCRIPTION
**Pull request by joshbodily**
* Incorporate cris's new artwork
* Add animations class
* Fixes a few issues
https://github.com/rmont881/EmojiWarsCC/issues/4
https://github.com/rmont881/EmojiWarsCC/issues/3

### CODE REVIEW

File: Classes/Collision.cpp [hunk 1]
```diff
@@ -36,13 +36,13 @@ namespace {
         if (dy <= 0.0)
             return results;
         
-        if (flags & 0x2 && x1 < x2)
+        if (flags.test(1) && x1 < x2)
             results.push_back(cocos2d::Vec2(dx, 0));
-        if (flags & 0x8 && x2 <= x2)
```
Comment (joshbodily) [inline Classes/Collision.cpp:41]: Oops => https://github.com/rmont881/EmojiWarsCC/issues/4

File: Classes/Character.cpp [hunk 2]
```diff
@@ -141,15 +142,16 @@ void Character::resolveCollision(
       _velocity.y = 0.0f;
       _onGround = true;
     }
-    if (closest.getNormalized().dot(cocos2d::Vec2::UNIT_Y) < 0.0f && _velocity.y < 0.0f) {
-      _velocity.y = 0.0f;
-    }
-    if (closest.getNormalized().dot(cocos2d::Vec2::UNIT_X) > 0.5f) {
+//    if (closest.getNormalized().dot(cocos2d::Vec2::UNIT_Y) < 0.0f && _velocity.y < 0.0f) {
```
Comment (joshbodily) [inline Classes/Character.cpp:145]: Disabling for now until I figure out a good criteria for stopping y velocity when character hits head on tile.

File: Classes/Collision.cpp [hunk 3]
```diff
@@ -36,13 +36,13 @@ namespace {
         if (dy <= 0.0)
             return results;
         
-        if (flags & 0x2 && x1 < x2)
+        if (flags.test(1) && x1 < x2)
```
Comment (joshbodily) [inline Classes/Collision.cpp:39]: numbers should be constants like COLLIDE_FLAG_LEFT instead of `1`, I'll fix later 😬

**Comment by rmont881** [comment]
:shipit:

**Pull request merged**